### PR TITLE
Game of Thrones javascript

### PR DESCRIPTION
**Pull request by ebwittenberg**

### CODE REVIEW

**Comment by ebwittenberg** [comment]
API request seems to work only some of the time

File: scripts/index.js [hunk 1]
```diff
@@ -1 +1,19 @@
...
+console.log(`There are ${characters.length} characters in the array.`);
+
+// How many characters names start with "A"?
+function charactersStartWithA(arr) {
+    let aCharacters = arr.filter(function(person) {
+        // include person if their name starts with a
+        return person.name[0] === 'A';
+    })
+    return aCharacters;
+}
```
Comment (CB987) [inline scripts/index.js:10]: 👍🏼

File: scripts/index.js [hunk 2]
```diff
@@ -1 +1,19 @@
...
+
+// How many characters start with "Z"?
+function charactersStartWithZ(arr) {
+    let zCharacters = arr.filter(function(person) {
+        // include person if their name starts with a
+        return person.name[0] === 'Z';
+    })
+    return zCharacters;
+}
```
Comment (CB987) [inline scripts/index.js]: 👍🏼

**Review by CB987 — APPROVED** [review]
nice!

File: scripts/index.js [hunk 3]
```diff
@@ -6,14 +6,34 @@ function charactersStartWithA(arr) {
...
+
+// How many characters are dead?
+function deadCharacters(arr) {
+    let deadCharacters = arr.filter(function(person) {
+        // include person if dead
+        return person.died;
+    })
+    return deadCharacters.length;
+}
```
Comment (CB987) [inline scripts/index.js:28]: 👍🏼

File: scripts/index.js [hunk 4]
```diff
@@ -6,14 +6,34 @@ function charactersStartWithA(arr) {
...
+
+// Who has the most titles?
+function mostTitles(arr) {
+    let mostTitles = arr.sort(function(characterA, characterB) {
+        return characterB.titles.length - characterA.titles.length;
+    })
+    return mostTitles[0].name;
+}
```
Comment (CB987) [inline scripts/index.js:36]: nice! I think this is the shortest one I've seen!

**Review by CB987 — APPROVED** [review]

File: scripts/index.js [hunk 5]
```diff
@@ -32,8 +32,36 @@ function mostTitles(arr) {
...
+console.log(mostTitles(characters));
 
+// How many are Valyrian?
+
+function valyrian(arr) {
+    let valyrianCharacters = arr.filter(person => {
+        return person.culture === 'Valyrian';
+    })
+    return valyrianCharacters;
+}
```
Comment (CB987) [inline scripts/index.js:45]: nice arrow function 😉

File: scripts/index.js [hunk 6]
```diff
@@ -32,8 +32,36 @@ function mostTitles(arr) {
...
+
+// Which actor plays "Hot Pie"?
+
+function findHotPie(arr) {
+    let hotPie = arr.filter(person => {
+        return person.name === 'Hot Pie';
+    })
+    return hotPie[0].playedBy[0];
+}
+console.log(findHotPie(characters));
```
Comment (CB987) [inline scripts/index.js]: 👌🏼

File: scripts/index.js [hunk 7]
```diff
@@ -32,8 +32,36 @@ function mostTitles(arr) {
...
+
+// How many characters are not in the TV show?
+
+function notInTvShow(arr) {
+    let notTv = arr.filter(person => {
+        return person.tvSeries.length > 1;
+    })
+    return notTv;
+}
```
Comment (CB987) [inline scripts/index.js:63]: 👍🏼

**Review by CB987 — APPROVED** [review]
nice clean, concise functions!

File: scripts/index.js [hunk 8]
```diff
@@ -64,4 +62,19 @@ function notInTvShow(arr) {
...
+function targaryens(arr) {
+
+    let targaryens = arr.filter(function(person) {
+        let firstLast = person.name.split(" ");
+        return firstLast[firstLast.length - 1] === 'Targaryen';
+    })
+    return targaryens;
+}
+
+console.log(targaryens(characters));
```
Comment (CB987) [inline scripts/index.js]: 👍🏼

**Review by CB987 — APPROVED** [review]

File: scripts/index.js [hunk 9]
```diff
@@ -73,8 +73,53 @@ function targaryens(arr) {
...
+            if (histogram[allegiance]) {
+                histogram[allegiance] += 1;
+            } else {
+                histogram[allegiance] = 1;
+            }
+        })
+    })
+    return histogram;
+}
+console.log(allegiancesClean(characters));
```
Comment (CB987) [inline scripts/index.js]: very nice! excellent refactoring.

**Review by CB987 — APPROVED** [review]

File: scripts/fetch-try.js [hunk 10]
```diff
@@ -0,0 +1,25 @@
+// create object with character name and allegiance
+function characterAllegiance(arr) {
+    let characterAllegiance = {};
+    arr.forEach(person => {
+        let allegianceArray = person.allegiances;
+        allegianceArray.forEach(allegiance => {
+            fetch(allegiance)
+                .then(function(response) {
+                    return response.json();
```
Comment (radishmouse) [inline scripts/fetch-try.js:9]: aw yeah!

File: scripts/fetch-try.js [hunk 11]
```diff
@@ -0,0 +1,25 @@
...
+                })
+                .then(function(myJson) {
+                    let result = myJson;
+                    if (characterAllegiance[result.name]) {
+                        characterAllegiance[result.name] += 1;
```
Comment (radishmouse) [inline scripts/fetch-try.js:14]: Excellent!

File: scripts/fetch-try.js [hunk 12]
```diff
@@ -0,0 +1,25 @@
...
+                    } else {
+                        characterAllegiance[result.name] = 1;
+                    }
+                })
+        })
+    })
+    return characterAllegiance;
```
Comment (radishmouse) [inline scripts/fetch-try.js:21]: One thing, though - you may have an incomplete histogram at the time the `console.log` runs. Each call to `fetch` may not finish before the histogram is returned.

**Review by radishmouse — APPROVED** [review]
Nice work going the extra mile with `fetch`!

**Pull request closed without merging**